x-pack/platform/packages/shared/onechat/onechat-genai-utils/tools/nl_search.ts

```diff
@@ -37,8 +41,10 @@ export const naturalLanguageSearch = async ({
     return { success: false, reason: 'No query was generated' };
   }
 
-  return await executeEsql({
+  const result = await executeEsql({
     query: generateResponse.queries[0],
     esClient,
   });
+
+  return { success: true, result };
```

Comment:
I'm, not super happy with this but we need a consistent return type for both success and failure scenario. Alternatively I can change the:

```
  if (generateResponse.queries.length < 1) {
    return { success: false, reason: 'No query was generated' };
  }
```

to

```
  if (generateResponse.queries.length < 1) {
    throw new Error('No query was generated');
  }  
```

That might actually be better...